src/Controllers/ElementalAreaController.php

```diff
@@ -29,23 +33,217 @@ class ElementalAreaController extends CMSMain
+            $this->jsonError(400);
+        }
+        $elementalArea = ElementalArea::get()->byID($elementalAreaID);
+        if (!$elementalArea) {
+            $this->jsonError(400);
+        }
+        if (!$elementalArea->canEdit()) {
+            $this->jsonError(403);
+        }
+        /** @var BaseElement $newElement */
+        $newElement = Injector::inst()->create($elementClass);
+        if (!$newElement->canCreate()) {
+            $this->jsonError(403);
+        }
+        // Assign the parent ID directly rather than via HasManyList to prevent multiple writes.
+        // See BaseElement::$has_one for the "Parent" naming.
+        try {
+            $newElement->ParentID = $elementalArea->ID;
+            $newElement->ensureSortSet();
+            if ($afterElementID !== null) {
+                /** @var ReorderElements $reorderer */
+                $reorderer = Injector::inst()->create(ReorderElements::class, $newElement);
+                $reorderer->reorder($afterElementID);
+            } else {
+                $newElement->write();
+            }
+            return $this->jsonSuccess(201);
+        } catch (Exception) {
+            $this->jsonError(500);
+        }
+    }
+
+    public function apiDuplicate(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $id = $this->getPostData()['id'] ?? '';
+         $element = BaseElement::get()->byID($id);
+        if (!$element) {
+            $this->jsonError(400);
+        }
+        if (!$element->canCreate()) {
+            $this->jsonError(403);
+        }
+        // check can edit the elemental area
+        $areaID = $element->ParentID;
+        $area = ElementalArea::get()->byID($areaID);
+        if (!$area) {
+            $this->jsonError(400);
+        }
+        if (!$area->canEdit()) {
+            $this->jsonError(403);
+        }
+        try {
+            // clone element
+            $clone = $element->duplicate(false);
+            $clone->Title = $this->getNewTitle($clone->Title ?? '');
+            $clone->Sort = 0; // must be zeroed for reorder to work
+            $area->Elements()->add($clone);
+            // reorder
+            $reorderer = Injector::inst()->create(ReorderElements::class, $clone);
+            $reorderer->reorder($id);
+            return $this->jsonSuccess(201);
+        } catch (Exception) {
+            $this->jsonError(500);
+        }
+    }
+
+    public function apiPublish(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $id = $this->getPostData()['id'] ?? '';
+        $element = BaseElement::get()->byID($id);
+        if (!$element) {
+            $this->jsonError(400);
+        }
+        if (!$element->canPublish()) {
+            $this->jsonError(403);
+        }
+        $element->publishRecursive();
+        return $this->jsonSuccess(204);
+    }
+
+    public function apiReadElementalArea(HTTPRequest $request): HTTPResponse
```

Comment:
Should maybe be `apiReadElements` and rename the route to match? We're not actually reading information about the area record, just about the elements inside it.

Reply:
Updated